### PR TITLE
Add Pull Request template

### PR DESCRIPTION
**Pull request by Berkyle**
#### Note: while reviewing this PR, think about what you'd want to see here that may not be present, or what is here that may be unnecessary.

# Description of Changes

Add `PULL_REQUEST_TEMPLATE.md` file. GitHub finds this file automatically once it is merged in to master; no additional configuration or settings need changed.

# Motivation & Context

Make a uniform pull request template.

# Type of change

- [ ] 🐛 Bug fix
- [ ] ✨ New feature/Enhancement
- [ ] 🛠 Technical currency
- [x] 💎 Code cleanliness

# Snapshots

#### This would usually be removed if there are no screenshots to share - see the comment in this area.

# Future considerations:

#### I see this section as giving the other reviewer more context on this code change. What does it stop short of accomplishing, so that it doesn't hit like, 1200 lines. For instance, I might say:
- Set up CI workflows via GitHub Actions.
- Set up a GitHub Pages configuration that automatically re-generates documentation when a new commit is merged into master.

# Notes
#### self-explanatory
For more information on Pull Request template, check out [this GitHub blog post](https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository#adding-a-pull-request-template)! Alright!

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+# Motivation & Context
+<!-- Why this code change was made. -->
+
+
+# Type of change
+<!-- Mark all that apply (the fewer the better) -->
+- [ ] 🐛 Bug fix
+- [ ] ✨ New feature
```
Comment (juzerzarif) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] ✨ New feature/Enhancement
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] 🛠 Technical currency
+- [ ] 💎 Code cleanliness
+
+# Snapshots
+<!-- If not applicable, remove this section. -->
+
+
+# Future considerations:
```
Comment (juzerzarif) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'd like to combine this and notes and just generalize them as Discussion. PRs would by and large be the way for us to talk about design choices and how they're gonna affect code changes, so an open ended area is really the only way to encapsulate everything.

Comment (Berkyle) [reply .github/PULL_REQUEST_TEMPLATE.md]: I see what you mean about setting a catch all "Discussion" section, but I do think we'd benefit from keeping a section that lays out next steps to be taken after each code change. Adding everything under "Discussion" feels too generic to me, but now that you mention it so does a "Notes" section...

What do you think about:
"Future considerations" => "Future Work" or "Next Steps", and 
"Notes" => "Discussion"

Comment (juzerzarif) [reply .github/PULL_REQUEST_TEMPLATE.md]: Talking about future work in a PR that doesn't actually do that work just crowds and detracts the conversation from the code change itself. I don't think a PR is the place for documenting next steps. You can set up something else for that (a google doc, trello board, etc.) but I don't go back to closed PRs to look at planned work.

Comment (Berkyle) [reply .github/PULL_REQUEST_TEMPLATE.md]: Gotcha, updated in 41181cd71308b1628c9b0b048eb0a58b703c5f3b. I think a doc or trello board would be useful, we can scrap that other one I made in like a night. I need to headsdown on the mobile app base but can make a new board if you don't have the time.

**Pull request merged**